QuietFactory.php

```diff
@@ -49,11 +48,21 @@ public function createItem($name, array $options = array())
         try {
             return $this->innerFactory->createItem($name, $options);
         } catch (RouteNotFoundException $e) {
-            $this->logger->error(sprintf('%s : %s', $name, $e->getMessage()));
+            if (null !== $this->logger) {
```

Comment:
what about instantiating a NullLogger in the constructor if the logger was null?